### PR TITLE
Adds timer measurements to reports.

### LINKED ISSUE #30: Reports should include time to DOMContentLoaded.

### LINKED ISSUE #31: Reports should include time to window.load.

### LINKED ISSUE #32: Reports should include time to first non-blank paint.

### LINKED ISSUE #34: Reports should include time to first user interaction.
One of `click`, `touch`, `keypress`, or `scroll`.

https://bugzilla.mozilla.org/show_bug.cgi?id=1307675

### LINKED ISSUE #38: Reports should include timestamp of request.

### PR DESCRIPTION
**Pull request by chuckharmston**
* Adds time to DOMContentLoaded to reports (closes #30).
* Adds time to window.load to reports (closes #31).
* Adds time to first paint to reports (closes #32).
* Adds time to first interaction to reports (closes #34).
* Adds timestamp to reports (closes #38).

### CODE REVIEW

File: src/data/pagemonitor.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+if (typeof window.pulse === 'undefined') {
+  window.pulse = {};
+}
+
+// Whenever the window receives the pulse-get-pagemonitor event, respond with
+// the full window.pulse object in a pulse-send-pagemonitor event.
+window.addEventListener('pulse-get-pagemonitor', () => {
+  var sendEvent = document.createEvent('CustomEvent');
```
Comment (pdehaan) [inline src/data/pagemonitor.js:8]: `var`? What is this, 2013?

File: src/data/timer-content-loaded.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const listener = document.addEventListener('pulse-send-pagemonitor', evt => {
+  self.port.emit('timer-content-loaded', evt.detail.timerContentLoaded);
+  document.removeEventListener('pulse-send-pagemonitor', listener);
+});
+
+var sendEvent = document.createEvent('CustomEvent');
```
Comment (pdehaan) [inline src/data/timer-content-loaded.js:6]: Similarly, not sure if we want to convert all these `var` to `const` below.

Relevant ESLint rules are `no-var` and `prefer-const`, if we want to add them to our ESLint config.

File: src/data/timer-first-interaction.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const listener = document.addEventListener('pulse-send-pagemonitor', evt => {
```
Comment (lmorchard) [inline src/data/timer-first-interaction.js:1]: nit: This module's overall pattern seems to repeat a lot. Might be nice to abstract into a shared utility function if possible. <strike>Can't remember if these scripts are getting run through webpack, though.</strike> [Looks like they aren't transformed](https://github.com/mozilla/pulse/blob/master/webpack.config.js#L48), so this is probably an impractical suggestion.

Comment (chuckharmston) [reply src/data/timer-first-interaction.js:1]: Yeah, that's exactly where I stand. Might go through and webpack them once they're all finished, but just trying to be practical right now.

**Review by lmorchard — APPROVED** [review]
Looks good to me, works locally. Tinkered a bit & tossed in some logging statements to see the timers getting populated. Can see them in the ping. :+1: I can merge, or can leave it in case you want to address nits.

**Pull request merged**